lib/lev/status.rb

```diff
@@ -56,18 +81,29 @@ def save(hash = {})
     end
 
     def add_error(error, options = { })
+      @errors ||= []
+
       options = { is_fatal: false }.merge(options)
-      push('errors', { is_fatal: options[:is_fatal],
-                       code: error.code,
-                       message: error.message })
+
+      @errors << { is_fatal: options[:is_fatal],
+                   code: error.code,
+                   message: error.message }
+
+      set(errors: @errors)
+    end
+
+    def as_json(options = {})
+      json = { id: id, status: status, progress: progress }
+      [options[:with]].flatten.each { |w| json[w] = send(w) }
```

Comment:
also, can you add a test for this `as_json` method?  Also FWIW this method returns a hash, not json.